server/src/routes/posts.ts

```diff
@@ -374,3 +375,44 @@ posts.delete("/:id", jwt, async (req: UserJwtRequest, res, next) => {
 		next(e)
 	}
 })
+
+posts.post(
+	"/import/gist/id/:id",
+	jwt,
+	celebrate({
+		body: {
+			visibility: Joi.string()
+				.custom(postVisibilitySchema, "valid visibility")
+				.required(),
+			password: Joi.string().optional(),
+			expiresAt: Joi.date().optional().allow(null, "")
+		}
+	}),
+	async (req: UserJwtRequest, res, next) => {
+		try {
+			const { id } = req.params
+			const { visibility, password, expiresAt } = req.body
+			const gist = await getGist(id)
+
+			let hashedPassword: string = ""
+			if (visibility === "protected") {
+				hashedPassword = crypto
```

Comment:
we have this `crypto.createHash...` line a few places. Would you mind refactoring it to a `lib/generate-password` or something? DRY and all that

Reply:
Actually, I wanted to restructure the whole routes into _thinner_ layers, and not cram domain logic, with DB with controller 😅. That's why I opted for the simplest (copy & paste) and discuss elsewhere how we could split the responsibilities. I created this proposal: https://github.com/MaxLeiter/Drift/issues/79